<launch>
	<node pkg="mpu9255" name="mpu9255_node" type="mpu9255_node" output="screen">
		<param name="print_min_max_mag" value="false"/> <!-- Prints the mag bias values -->
		<param name="print_min_max_gyro" value="false"/> <!-- For checking the range of accel -->
		<param name="print_min_max_acc" value="false"/> <!-- For checking the range of accel -->
		<param name="print_rolling_mean_acc" value="false"/> <!-- For checking the stationary bias of accel -->
		
		<param name="imu_covar" value="0.01"/>
		<param name="mag_covar" value="0.01"/>
		
		<param name="acc_bias_x" value="1.273715"/> <!-- 1.0605 -->
		<param name="acc_bias_y" value="-0.237026"/> <!-- 0.2486 -->
		<param name="acc_bias_z" value="0.800197"/> <!-- 0.7708 -->
		
		<param name="gyr_bias_x" value="0.017825"/>
		<param name="gyr_bias_y" value="0.010110"/>
		<param name="gyr_bias_z" value="0.000266"/>
	</node>

	<node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter_node" output="screen">
		<param name="gain" value="0.5"/>
		<!-- <param name="zeta" value="-0.01"/> -->
		<param name="orientation_stddev" value="0.1"/>
		 
		<param name="mag_bias_x" value="0.000009"/>  <!--  -0.000009  -->
		<param name="mag_bias_y" value="-0.000004"/>  <!--   0.000005   -->
		<param name="mag_bias_z" value="-0.000022"/>  <!--   0.000022  -->
		<param name="world_frame" value="enu"/>
		<param name="fixed_frame" value="base_link"/>
		<param name="use_magnetic_field_msg" value="true"/> 
 		<param name="publish_tf"    value="false"/>  <!-- Only use this for debug! -->
		<param name="remove_gravity_vector" value="false"/> <!-- This is done by robot_localization -->
	</node>

</launch>
